### PR TITLE
In nin array subdoc

### PR DESCRIPTION
**Pull request by Yuqi-Du**
**What this PR does**:
support $in and $nin for array and sub_doc
```
{
  "find": {
   "filter" : {"array_field": {"$in" : [["A", "B", "C"], ["X", "Y", "Z"]] } }
  }
}
{
  "find": {
   "filter" : {"sub_doc_field": {"$in" : [ { "A" : "B" }] } }
  }
}
```

**Which issue(s) this PR fixes**:
Fixes [732](https://github.com/stargate/jsonapi/issues/732)

**Checklist**
- [x] Changes manually tested
- [x] Automated Tests added/updated
- [x] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

**Review by vkarpov15 — APPROVED** [review]
One minor comment but LGTM overall

File: src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/DBFilterBase.java [hunk 1]
```diff
@@ -411,25 +411,59 @@ public List<BuiltCondition> getAll() {
...
+          final ArrayList<BuiltCondition> inResult = new ArrayList<>();
+          for (Object value : values) {
+            if (value.getClass().equals(LinkedHashMap.class)) {
+              // array element is sub_doc
+              inResult.add(
+                  BuiltCondition.of(
+                      BuiltCondition.LHS.mapAccess("query_text_values", Values.NULL),
+                      Predicate.EQ,
+                      new JsonTerm(this.getPath(), getHash(new DocValueHasher(), value))));
+            } else if (value.getClass().equals(ArrayList.class)) {
```
Comment (tatu-at-datastax) [inline src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/DBFilterBase.java:423]: Similarly here

    if (value instanceof List)

File: src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/DBFilterBase.java [hunk 2]
```diff
@@ -411,25 +411,59 @@ public List<BuiltCondition> getAll() {
...
-              .map(
-                  v ->
-                      BuiltCondition.of(
-                          DATA_CONTAINS,
-                          Predicate.CONTAINS,
-                          new JsonTerm(getHashValue(new DocValueHasher(), getPath(), v))))
-              .collect(Collectors.toList());
+          final ArrayList<BuiltCondition> inResult = new ArrayList<>();
+          for (Object value : values) {
+            if (value.getClass().equals(LinkedHashMap.class)) {
```
[the remark this answers is not in the document]
Comment (Yuqi-Du) [reply src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/DBFilterBase.java:416]: got it

File: src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/DBFilterBase.java (hunk 1 above)
Comment (Yuqi-Du) [reply src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/DBFilterBase.java:423]: thanks

File: src/test/java/io/stargate/sgv2/jsonapi/api/v1/InAndNinIntegrationTest.java [hunk 3]
```diff
@@ -576,18 +639,46 @@ public void nonIdNinEmptyArray() {
...
+          .body(json)
+          .when()
+          .post(CollectionResource.BASE_PATH, namespaceName, collectionName)
+          .then()
+          .statusCode(200)
+          // except doc 5
+          .body("data.documents", hasSize(5))
+          .body("status", is(nullValue()))
+          .body("errors", is(nullValue()))
+          .body("data.documents", notNullValue());
```
[the remark this answers is not in the document]
Comment (Yuqi-Du) [reply src/test/java/io/stargate/sgv2/jsonapi/api/v1/InAndNinIntegrationTest.java:672]: thanks, will add these.

File: src/test/java/io/stargate/sgv2/jsonapi/api/v1/InAndNinIntegrationTest.java [hunk 4]
```diff
@@ -576,18 +639,46 @@ public void nonIdNinEmptyArray() {
...
           .then()
           .statusCode(200)
-          .body("data.documents", hasSize(6))
+          // except doc 5
+          .body("data.documents", hasSize(5))
           .body("status", is(nullValue()))
-          .body("errors", is(nullValue()));
+          .body("errors", is(nullValue()))
+          .body("data.documents", notNullValue());
     }
```
Comment (Hazel-Datastax) [inline src/test/java/io/stargate/sgv2/jsonapi/api/v1/InAndNinIntegrationTest.java:647]: Why this test doesn't have the expected value? Perhaps you could add `.body("data.documents", hasSize(5))` if you don't want to list them all.

**Review by Hazel-Datastax — APPROVED** [review]
LGTM

File: src/test/java/io/stargate/sgv2/jsonapi/api/v1/InAndNinIntegrationTest.java (hunk 4 above)
Comment (Yuqi-Du) [reply src/test/java/io/stargate/sgv2/jsonapi/api/v1/InAndNinIntegrationTest.java:688]: thanks fixed.

**Review by vkarpov15 — COMMENTED** [review]
LGTM

**Pull request merged**